vtkKNNGraphFilter.cxx

```diff
@@ -125,9 +126,17 @@ int vtkKNNGraphFilter::RequestData(vtkInformation *vtkNotUsed(request),
         }
       }
     }
-  
+
+  double normal[3] = {0, 0, 0};
+  vtkSmartPointer<vtkPoints> normals = vtkSmartPointer<vtkPoints>::New();
+  for (unsigned int i = 0; i < input->GetNumberOfPoints(); i++)
+  {	
+	input->GetPointData()->GetNormals()->GetTuple(i, normal);
+	normals->InsertNextPoint(normal);
+  }	
+
   //add the point coordinates to the graph
-  g->SetPoints(input->GetPoints());
+  g->SetPoints(normals);
```

Comment:
Here again the normals are being interpreted as points? This filter really should not have anything to do with normals at all right? It should just compute some connectivity structure over the points (based on connecting the k-nearest neighbors of each point).